### ISSUE TITLE
How can I set a DevelopmentTeam in TargetAttributes?

### ISSUE THREAD

**Issue by mosesoak** [CocoaPods/Xcodeproj#384]
I've been hunting around but can't figure out how to set an arbitrary attribute on the project.

Specifically, a fresh XCode project with no "Team" setting is missing an entry for the target in its TargetAttributes object. I need to create an object and add it into the attribute.

The 'attributes' property is simply a hash, so how would I do this? Any help is much appreciated, thanks.

**Comment by neonichu** [comment]
I don't think this is supported at all at the moment as the `Team` setting was just introduced with Xcode 8.

**Comment by mosesoak** [comment]
Hmm - not sure we're talking about the same thing. I'll give you a clearer description.

In Xcode Project Settings -> Target -> General,

![screen shot 2016-06-27 at 2 20 30 pm](https://cloud.githubusercontent.com/assets/24345/16396185/6487c9d6-3c72-11e6-9e26-e506568fc937.png)

So if you've selected a Team using this dropdown, Xcode modifies its pbxproj with a `DevelopmentTeam` in the project's `TargetAttributes`, keyed from the target UUID. However, if you haven't changed that dropdown the UUID entry is missing, so I have to generate it manually. 

Originally I had tried using Fastlane's update_project_team action, but it fails since their script doesn't generate the UUID entry if it doesn't exist yet – note that they're also resorting to find/replace on the project file for this. (I submitted an issue to them here https://github.com/fastlane/fastlane/issues/5109)

My workaround right now is

```ruby
    # Team
    main_target = project.targets.find { |t| t.name == project_name }
    main_target_uuid = main_target && main_target.uuid
    if main_target_uuid
      target_atts_obj = project.root_object.attributes['TargetAttributes']
      target_atts_obj[main_target_uuid] ||= {}
      target_atts_obj.each do |key, val|
        val['DevelopmentTeam'] = $team_id
      end
      replace_pattern_in_file(pbxproj_path, /(^\s*?)(TargetAttributes.*)(\1\};)/m, "#{JSON.pretty_generate(target_atts_obj)};")
    end
```

**Comment by neonichu** [comment]
Sorry, I think my comment wasn't clear, I meant that I don't think this is supported by Xcodeproj at the moment, because it was just introduced with Xcode 8 :)

**Comment by mosesoak** [comment]
okay cool, good to know that it might be possible soon then! Thanks

**Comment by aidansteele** [comment]
@neonichu For what it's worth, this attribute exists in Xcode 7.3.1 (what I have installed right now - it probably existed earlier as well). It just seems to be _mandatory_ as of Xcode 8. I'm not sure if that helps?

(As an irrelevant aside, I love your work - keep it up!)

**Comment by mosesoak** [comment]
Thanks! Oh so, you're saying we won't encounter the problem where the attribute doesn't exist at all before Team has been set? That would be nice because it would fix Fastlane's existing command.

**Comment by AsTryE** [comment]
xcode7.3
TargetAttributes = {
					88E9EC28B8B46C3631E6B242B50F4442 = {
						DevelopmentTeam = ***;
					};
					D991909832991B04B9E3D1C11E4BCA60 = {
						DevelopmentTeam = ***;
					};
				};
}
me too,how to change TargetAttributes ? becauser i temporarily not upgraded to Xcode8....thanks.

**Comment by yakimant** [comment]
Hi everyone!

Is it still not possible?
